api/v1beta1/common.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/endpoint"
+)
+
+// GlanceAPITemplate defines the desired state of GlanceAPI
+type GlanceAPITemplate struct {
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=glance
+	// ServiceUser - optional username used for this service to register in glance
+	ServiceUser string `json:"serviceUser"`
+
+	// +kubebuilder:validation:Required
+	// GlanceAPI Container Image URL
+	ContainerImage string `json:"containerImage"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=internal;external
+	// +kubebuilder:default=external
+	APIType string `json:"apiType"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Maximum=32
+	// +kubebuilder:validation:Minimum=0
+	// Replicas of glance API to run
+	Replicas int32 `json:"replicas"`
+
+	// +kubebuilder:validation:Optional
+	// DatabaseHostname - Glance Database Hostname
+	DatabaseHostname string `json:"databaseHostname"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=glance
+	// DatabaseUser - optional username used for glance DB, defaults to glance
+	// TODO: -> implement needs work in mariadb-operator, right now only glance
+	DatabaseUser string `json:"databaseUser"`
```

Comment:
I guess both the internal and the external GlanceAPI uses the same DB schema with the same user / password. If so then this can be removed here and the value in GlanceAPI.Spec.DatabaseUser can be filled from Glance.Spec.DatabaseUser automatically.